src/transformers/widget.ts

```diff
@@ -2,16 +2,40 @@ import { Bot } from "../bot.ts";
 import { GuildWidget } from "../types/guilds/guildWidget.ts";
 import { SnakeCasedPropertiesDeep } from "../types/util.ts";
 
+import { DiscordenoChannel } from "./channel.ts";
+import { DiscordenoMember } from "./member.ts";
+
 export function transformWidget(bot: Bot, payload: SnakeCasedPropertiesDeep<GuildWidget>): DiscordenoWidget {
   return {
-    channelId: payload.channel_id ? bot.transformers.snowflake(payload.channel_id) : undefined,
-    enabled: payload.enabled,
+    id: bot.transformers.snowflake(payload.id),
+    name: payload.name,
+    instantInvite: payload.instant_invite,
+    channels: payload.channels.map((channel) => ({
+      id: channel.id ? bot.transformers.snowflake(channel.id) : undefined,
+      name: channel.name,
+      position: channel.position,
+    })),
+    members: payload.members.map((member) => ({
+      id: member.user?.id ? bot.transformers.snowflake(member.user.id) : undefined,
+      username: member.user?.username,
+      discriminator: Number(member.user?.discriminator),
+      avatar: member.avatar ? bot.utils.iconHashToBigInt(member.avatar) : undefined,
+    })),
+    presenceCount: payload.presence_count,
   };
 }
 
-export interface DiscordenoWidget {
-  /** Whether the widget is enabled */
-  enabled: boolean;
-  /** The widget channel id */
-  channelId?: bigint;
+interface DiscordenoWidget {
+  /** guild id */
+  id: bigint;
+  /** guild name (2-100 characters) */
+  name: string;
+  /** instant invite for the guilds specified widget invite channel */
+  instantInvite: string | null;
+  /** voice and stage channels which are accessible by @everyone */
+  channels: Partial<DiscordenoChannel>[];
+  /** special widget user objects that includes users presence (Limit 100) */
+  members: Partial<DiscordenoMember>[];
```

Comment:
The types of channel and members should be made more acurate. Blame Discord for always using Partial instead of exact definitions .-.

This can be taken as a reference although it does might not be 100% accurate so please confirm it with testing.
![image](https://user-images.githubusercontent.com/72305210/152604755-0c56251a-b2b9-4945-858b-7cc92984f08d.png)

*The same goes for the duped types.*